core/src/test/java/fj/control/db/TestDbState.java

```diff
@@ -0,0 +1,53 @@
+package fj.control.db;
+
+import fj.Unit;
+import fj.data.Option;
+import org.junit.Test;
+
+import java.sql.*;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+public class TestDbState {
+    @Test
+    public void testWriter() throws SQLException {
+        int TEN = 10;
+        DbState writer = DbState.writer(DbState.driverManager("jdbc:h2:mem:"));
+
+        DB<Unit> setup = new DB<Unit>() {
+            @Override
+            public Unit run(Connection c) throws SQLException {
```

Comment:
I see, the db method takes a function (no exception), but the run method can throw an exception.  We probably want to pass in a Try1 instead.